packages/datadog-plugin-http/test/client.spec.js

```diff
@@ -804,6 +805,103 @@ describe('Plugin', () => {
           })
         })
 
+        if (satisfies(process.version, '>=20')) {
+          it('should not record default HTTP agent timeout as error with Node 20', done => {
+            const app = express()
+
+            app.get('/user', async (req, res) => {
+              await new Promise(resolve => {
+                setTimeout(resolve, 6 * 1000) // over 5s default
```

Comment:
Can you search the codebase for usage of `clock.tick()` and try to use it here?

In the current state these tests will add 6s of time, but clock.tick might be able to expedite that.

I know it works with setTimeout. I don't know if it works with the inner socket timeout stuff (like, is that handled in C?)

Reply:
Tried changing it to 

```javascript
app.get('/user', async (req, res) => {
  clock.tick(6000)
  res.status(200).send()
})
```
For the test I added that checks custom agent timeouts still attach errors, and it failed because the test timed out without ever passing (so the error wasn't attached) - I'm guessing the inner socket timeout clocks don't play nicely with this? It looks like the timeout event didn't trigger.

Reply:
I guess there's no real way around it then...

I suspect these 3 tests for each of 2 modules (http, https) running at 6s will increase test run time by ~36s.